google-cloud-storage/src/test/java/com/google/cloud/storage/it/StorageNativeCanary.java

```diff
@@ -0,0 +1,187 @@
+            .setStorage(storage)
+            .setBucketInfo(BucketInfo.of("java-storage-grpc-" + UUID.randomUUID()))
+            .build()) {
+      String bucketName = tempB.getBucket().getName();
+      String obj1Name = UUID.randomUUID().toString();
+      String obj2Name = UUID.randomUUID().toString();
+
+      // insert 2 objects
+      BlobInfo info1 = BlobInfo.newBuilder(bucketName, obj1Name).build();
+      BlobInfo info2 = BlobInfo.newBuilder(bucketName, obj2Name).build();
+      uploadUsingWriter(storage, info1);
+      uploadUsingWriter(storage, info2);
+
+      // list objects
+      Page<Blob> page = storage.list(bucketName, BlobListOption.pageSize(1));
+      List<Blob> blobs = ImmutableList.copyOf(page.iterateAll());
+
+      // read all bytes of each object
+      List<BlobWithContent> actual =
+          blobs.stream()
+              .map(info -> readAll(storage, info))
+              .collect(ImmutableList.toImmutableList());
+
+      List<Boolean> deletes =
+          blobs.stream()
+              .map(b -> storage.delete(b.getBlobId(), BlobSourceOption.generationMatch()))
+              .collect(ImmutableList.toImmutableList());
+
+      assertAll(
+          () -> {
+            List<String> actualNames =
+                actual.stream()
+                    .map(BlobWithContent::getInfo)
+                    .map(BlobInfo::getBlobId)
+                    .map(BlobId::getName)
+                    .collect(ImmutableList.toImmutableList());
+
+            assertThat(actualNames).containsExactly(info1.getName(), info2.getName());
+          },
+          () -> {
+            // if the content isn't equal carry it through
+            List<BlobWithContent> contentNotEquals =
+                actual.stream()
+                    .filter(bwc -> !Arrays.equals(bwc.getContent(), bytes))
+                    .collect(ImmutableList.toImmutableList());
+            assertThat(contentNotEquals).isEmpty();
+          },
+          () -> assertThat(deletes.stream().anyMatch(isFalse())).isFalse());
+    }
+  }
+
+  private static void uploadUsingWriter(Storage storage, BlobInfo info) throws IOException {
+    try (WriteChannel w = storage.writer(info, BlobWriteOption.doesNotExist())) {
+      // set our size to the smallest resumable size, so we can send multiple requests
+      w.setChunkSize(_256KiB);
+      ByteStreams.copy(Channels.newChannel(new ByteArrayInputStream(bytes)), w);
+    }
+  }
+
+  private static BlobWithContent readAll(Storage storage, BlobInfo info) {
+    try (ReadChannel r = storage.reader(info.getBlobId(), BlobSourceOption.generationMatch());
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        WritableByteChannel w = Channels.newChannel(baos)) {
+      // only buffer up to half the object
+      r.setChunkSize(_256KiB);
+      ByteStreams.copy(r, w);
+      return new BlobWithContent(info, baos.toByteArray());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
Could we make the thrown exception a little more specific or perhaps add a `throws IOException` to the method signature?  This could help with debugging and also prevent the caller of this method from catching unrelated exceptions since a`RuntimeException` can happen for a number of reasons. 

https://effectivejava.googleplex.com/third-edition#page=323

Reply:
In this case the RuntimeException wrapping is necessary because this method is called within a lambda which can have a checked exception. The fact that this wrapping is taking place for a private class in a private utility that doesn't escape this class I'm not sure a custom exception type provides much value, but I can change it.

Reply:
Hm I see. How about we do something like `RuntimeException('couldn't read foo', e)` instead? where the message gives some context into where the try block went wrong.

Reply:
We're carrying the cause through to be the cause of the runtime exception, so any exception that does pop out will have the original stacktrace with it. I've changed the thrown exception to be an internal `RuntimeIOException` to disambiguate from some other RuntimeException

Reply:
Got it! That makes sense, thanks for the context!